annexes/init-scripts-and-daemonize.md

```diff
@@ -0,0 +1,242 @@
+    start-stop-daemon --start --quiet --make-pidfile --pidfile $PIDFILE --exec $DAEMON --name $NAME --background -- $DAEMON_OPTS || return 2
+}
+
+resetconf() {
+    rm -rf $FB_DATABASE
+}
+
+do_stop()
+{
+    start-stop-daemon --stop --quiet --remove-pidfile --retry=TERM/30/KILL/5 --pidfile $PIDFILE --name $NAME
+    RETVAL="$?"
+    [ "$RETVAL" = 2 ] && return 2
+    start-stop-daemon --stop --quiet --oknodo --retry=0/30/KILL/5 --exec $DAEMON
+    [ "$?" = 2 ] && return 2
+    rm -f $PIDFILE
+    return "$RETVAL"
+}
+
+case "$1" in
+  start)
+    [ "$VERBOSE" != no ] && log_daemon_msg "Starting $DESC" "$NAME"
+    do_start
+    case "$?" in
+	0|1) [ "$VERBOSE" != no ] && log_end_msg 0 ;;
+	2) [ "$VERBOSE" != no ] && log_end_msg 1 ;;
+    esac
+    ;;
+  stop)
+    [ "$VERBOSE" != no ] && log_daemon_msg "Stopping $DESC" "$NAME"
+    do_stop
+    case "$?" in
+	0|1) [ "$VERBOSE" != no ] && log_end_msg 0 ;;
+	2) [ "$VERBOSE" != no ] && log_end_msg 1 ;;
+    esac
+    ;;
+  status)
+    status_of_proc "$DAEMON" "$NAME" && exit 0 || exit $?
+    ;;
+  restart|force-reload)
+    log_daemon_msg "Restarting $DESC" "$NAME"
+    do_stop
+    case "$?" in
+      0)
+	do_start
+	case "$?" in
+	    0) log_end_msg 0 ;;
+	esac
+	;;
+      *)
+	log_end_msg 1
+	;;
+    esac
+    ;;
+  resetconf)
+    log_daemon_msg "Resetting/erasing all and make new config instance of $DESC" "$NAME"
+    do_stop
+    case "$?" in
+      0)
+	resetconf
+	do_start
+	case "$?" in
+	    0) log_end_msg 0 ;;
+	esac
+	;;
+      *)
+	log_end_msg 1
+	;;
+    esac
+    ;;
+  *)
+    echo "Usage: $SCRIPTNAME {start|stop|status|restart|resetconf}" >&2
+    exit 3
+    ;;
+esac
+
+```
+
+## OpenRC
+
+This init script can be used on Alpine and Gentoo based systems like Funtoo, for others openrc style you can manage the start and stop commands.
+
+The configuration unfortunatelly cannot be managed by using a `/etc/conf.d/filebrowser` file yet with the environment variables of `FB_`:
+
+```bash
+#!/sbin/openrc-run
+# /usr/local must be adapted to installed binary
+# Author: PICCORO Lenz McKAY <mckaygerhard@gmail.com>
+
+name=$RC_SVCNAME
+cfgfile="/etc/$RC_SVCNAME/$RC_SVCNAME.conf"
+command="/usr/local/filebrowser"
+command_args="--database /srv/$RC_SVCNAME.db"
+FBPID="/run/$RC_SVCNAME.pid"
+pidfile="${FBPID}"
+command_background="yes"
+extra_started_commands="resetconf"
+
+checkconfig() {
+    if [ -z "${FBPID}" ] ; then
+	FBPID="/run/filebrowser.pid"
+    fi
+    if [ -z "${FB_ROOT}" ] ; then
+	FB_ROOT="/srv"
+    fi
+}
+
+start() {
+    checkconfig || return 1
+    ebegin "Starting ${SVCNAME}"
+    start-stop-daemon --start --quiet --exec /usr/sbin/lighttpd --pidfile "${FBPID}" --make-pidfile -- 
+    eend $?
+}
+
+stop() {
+    local rv=0
+    ebegin "Stopping ${SVCNAME}"
+    start-stop-daemon --stop --quiet --pidfile "${FBPID}"
+    eend $?
+}
+
+resetconf() {
+    stop
+    checkconfig || return 1
+    ebegin "Reseting/erasing all settings files not implemented. help here"
+}
```

Comment:
I don't think we need a reset functionality in general

Reply:
this is, cos, sometimes the system due to sudden shutdown, due to an error in the same application, closes without synchronizing with the script, **because as you will notice the program does not create its own pid**, that is why that step is here

Reply:
Init script shouldn't destroy a database or any configuration file

Reply:
ok will be removed.. i need to fast forward so many complaints.. puff